here-naksha-app-service/src/test/java/com/here/naksha/app/service/UpdateFeatureTestHelper.java

```diff
@@ -0,0 +1,308 @@
+        final HttpResponse<String> response =
+                nakshaClient.put("hub/spaces/" + space.getId() + "/features/wrong-id", bodyJson, streamId);
+
+        // Then: Perform assertions
+        assertEquals(409, response.statusCode(), "ResCode mismatch");
+        JSONAssert.assertEquals(
+                "Update Feature error response doesn't match",
+                expectedBodyPart,
+                response.body(),
+                JSONCompareMode.LENIENT);
+        assertEquals(streamId, getHeader(response, HDR_STREAM_ID), "StreamId mismatch");
+    }
+
+    void tc0503_testUpdateFeatureWithMismatchingId() throws Exception {
+        // Test API : PUT /hub/spaces/{spaceId}/features/{featureId}
+
+        // Read request body
+        final String bodyJson = loadFileOrFail("TC0502_updateFeatureWithWrongUriId/request.json");
+        // TODO: include geometry after Cursor-related changes ->
+        final Space space = parseJsonFileOrFail("TC0500_updateFeatures/create_space.json", Space.class);
+        final String expectedBodyPart = loadFileOrFail("TC0503_updateFeatureMismatchingId/response.json");
+        final String streamId = UUID.randomUUID().toString();
+
+        // When: Create Features request is submitted to NakshaHub Space Storage instance
+        final HttpResponse<String> response =
+                nakshaClient.put("hub/spaces/" + space.getId() + "/features/my-custom-id-301-1", bodyJson, streamId);
+
+        // Then: Perform assertions
+        assertEquals(400, response.statusCode(), "ResCode mismatch");
+        JSONAssert.assertEquals(
+                "Update Feature error response doesn't match",
+                expectedBodyPart,
+                response.body(),
+                JSONCompareMode.LENIENT);
+        assertEquals(streamId, getHeader(response, HDR_STREAM_ID), "StreamId mismatch");
+    }
+
+    void tc0504_testUpdateFeaturesNoId() throws Exception {
+        // Test API : PUT /hub/spaces/{spaceId}/features
+        final String streamId = UUID.randomUUID().toString();
+
+        // Read request body
+        final String bodyJson = loadFileOrFail("TC0504_updateFeaturesNoIds/request.json");
+        // TODO: include geometry after Cursor-related changes ->
+        final Space space = parseJsonFileOrFail("TC0500_updateFeatures/create_space.json", Space.class);
+        // When: Create Features request is submitted to NakshaHub Space Storage instance
+        final HttpResponse<String> response =
+                nakshaClient.put("hub/spaces/" + space.getId() + "/features", bodyJson, streamId);
+
+        // Then: Perform assertions
+        assertEquals(200, response.statusCode(), "ResCode mismatch");
+        assertEquals(streamId, getHeader(response, HDR_STREAM_ID), "StreamId mismatch");
+
+        final XyzFeatureCollection responseFeatureCollection = parseJson(response.body(), XyzFeatureCollection.class);
```

Comment:
This custom `inserted` list check can be better replaced by having function similar to `CreateFeatureTestHelper.additionalCustomAssertions()`, except that there is no need of `prefixId` as an argument here.